bin/du.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python3
+# Copyright (C) 2018 Evgeny Golyshev <eugulixes@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# The main motivation to create a substitution for du is that it can sometimes
+# provide inaccurate result. For example, directory size may vary depending on
+# when du is called -- before or after transferring the directory to an image.
+
+import os
+import sys
+from optparse import OptionParser
+
+
+def walk(path):
+    """Recursively yields DirEntry objects for given directory. """
+    for entry in os.scandir(path):
+        if entry.is_dir(follow_symlinks=False):
```

Comment:
This method can raise OSError, such as PermissionError, but FileNotFoundError is caught and not raised.